### ISSUE TITLE
Issue with unused variables and exported

### ISSUE THREAD

**Issue by rhernandog** [eslint/eslint#8159]
**Tell us about your environment**

* **ESLint Version:3.16.1**
* **Node Version:6.9.2**
* **npm Version:3.10.9**

**What parser (default, Babel-ESLint, etc.) are you using?**
Default with VS Code extension [(Link)](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)

**Please show your full configuration:**

```

{
	"env": {
		"browser": true,
		"commonjs": true,
		"es6": true,
		"node": true
	},
	"parserOptions": {
		"ecmaVersion": 6,
		"sourceType": "module",
		"ecmaFeatures": {
			"jsx": true,
			"classes": true,
			"defaultParams": true,
			"experimentalObjectRestSpread": true
		}
	},
	"rules": {
		"no-const-assign": 1,
		"no-extra-semi": 0,
		"semi": 0,
		"no-fallthrough": 0,
		"no-empty": 0,
		"no-mixed-spaces-and-tabs": 0,
		"no-redeclare": 0,
		"no-this-before-super": 1,
		"no-undef": 1,
		"no-unreachable": 1,
		"no-use-before-define": 0,
		"constructor-super": 1,
		"curly": 0,
		"eqeqeq": 0,
		"func-names": 0,
		"valid-typeof": 1,
		"indent": [1, "tab"],
		"no-unused-vars": 1
	}
}

```

**What did you do? Please include the actual source code causing the issue.**

```js
/* exported React */
/* globals TweenLite */
import React, { Component } from "react";
```

**What did you expect to happen?**
No warning regarding the fact that I'm importing React but not using it and I'm explicitly indicating that is an exported variable.

**What actually happened? Please include the actual, raw output from ESLint.**

file: '/src/modules/components/modal.jsx'
severity: 'Warning'
message: ''React' is defined but never used. (no-unused-vars)'
at: '3,8'
source: 'eslint'

---
On a related issue I experienced the same issue while working on a project some time ago with several ES5 files bundled together with Browserify. One of the files contained all the variables for the plugin. This file was then required into all the other files and every variable that wasn't used on the variables file returned the same warning, so I ended up grouping them and using `/* eslint-disable */` and after all of them `/* eslint-enable */` to hide all the warnings.

**Comment by alberto** [comment]
`exported` has no effect in ES modules. See http://eslint.org/docs/rules/no-unused-vars#exported

**Comment by rhernandog** [comment]
So I guess the question is, there is a way to avoid this on ES modules?

**Comment by rhernandog** [comment]
Hi,

I added this to the code:
```js
let test;
```
Then run `eslint modal.jsx` on the command line and returned the same error:
```
14:5  warning  'test' is defined but never used  no-unused-vars
```

**Comment by alberto** [comment]
You could use `{ "varsIgnorePattern": "React" }`

**Comment by alberto** [comment]
Closing this issue as it looks like the question has been answered. Please feel free to visit us in the ESLint Gitter if you have any other issues!
